src/pipeline.cxx

```diff
@@ -261,6 +261,10 @@ bool pqxx::pipeline::obtain_result(bool expect_none)
 
 void pqxx::pipeline::obtain_dummy()
 {
+  // Allocate once, re-use across invocations.
+  static auto const text{
+    std::make_shared<std::string>("[DUMMY PIPELINE QUERY]")};
```

Comment:
I'm honestly really confused by this.

Why is the query text represented as a shared_ptr to string?

Reply:
To make it cheap to copy, without requiring another level of indirection between the `pqxx::result` object and its `PGresult`.